packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt

```diff
@@ -0,0 +1,305 @@
+package expo.modules.imagepicker
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.media.MediaMetadataRetriever
+import android.net.Uri
+import android.os.Bundle
+import android.provider.MediaStore
+import android.util.Log
+import com.theartofdev.edmodo.cropper.CropImage
+import expo.modules.imagepicker.ImagePickerOptions.Companion.optionsFromMap
+import expo.modules.imagepicker.exporters.CompressionImageExporter
+import expo.modules.imagepicker.exporters.CropImageExporter
+import expo.modules.imagepicker.exporters.ImageExporter
+import expo.modules.imagepicker.exporters.RawImageExporter
+import expo.modules.imagepicker.fileproviders.CacheFileProvider
+import expo.modules.imagepicker.fileproviders.CropFileProvider
+import expo.modules.imagepicker.tasks.ImageResultTask
+import expo.modules.imagepicker.tasks.VideoResultTask
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ActivityEventListener
+import org.unimodules.core.interfaces.ActivityProvider
+import org.unimodules.core.interfaces.ExpoMethod
+import org.unimodules.core.interfaces.services.UIManager
+import org.unimodules.core.utilities.FileUtilities.generateOutputPath
+import org.unimodules.interfaces.imageloader.ImageLoader
+import org.unimodules.interfaces.permissions.Permissions
+import org.unimodules.interfaces.permissions.PermissionsResponse
+import org.unimodules.interfaces.permissions.PermissionsResponseListener
+import org.unimodules.interfaces.permissions.PermissionsStatus
+import java.io.IOException
+import java.lang.ref.WeakReference
+
+class ImagePickerModule(private val mContext: Context,
+                        val moduleRegistryPropertyDelegate: ModuleRegistryPropertyDelegate = ModuleRegistryPropertyDelegate())
+  : ExportedModule(mContext), ActivityEventListener {
+
+  private var mCameraCaptureURI: Uri? = null
+  private var mPromise: Promise? = null
+  private var mPickerOptions: ImagePickerOptions? = null
+
+  private val mExperienceActivity: WeakReference<Activity> by moduleRegistry { activityProvider: ActivityProvider ->
+    WeakReference(activityProvider.currentActivity)
+  }
+  private val mImageLoader: ImageLoader by moduleRegistry()
+  private val mUIManager: UIManager by moduleRegistry()
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    moduleRegistryPropertyDelegate.onCreate(moduleRegistry)
+  }
+
+  private fun getModuleRegistry() = moduleRegistryPropertyDelegate.moduleRegistry!!
```

Comment:
I assume that this `propertyDelegation` couldn't help us with this one?
It would be wonderful if we could go with sth like this 😅 
```kotlin
private val mModuleRegistry: ModuleRegistry by moduleRegistry
```